src/platform/i386/lapic.c

```diff
@@ -86,26 +87,35 @@ lapic_read_reg(u32_t off)
 
 void
 lapic_set_timer(int timer_type, cycles_t deadline)
-{	
-	u32_t high, low;
-	int retries = RETRY_ITERS;
+{
+	u64_t now;
 
-	if (timer_type != TSC_DEADLINE) {
-		printk("Mode not supported\n");
-		assert(0);
-	}
+	assert (timer_type != PERIODIC);
+
+	rdtscll(now);
+	if (deadline <= now) return;
+
+	if (timer_type == ONESHOT) {
+
+		lapic_write_reg(LAPIC_INIT_COUNT_REG, (u32_t)(deadline - now));
```

Comment:
Does this really work?  I thought the granularity of the oneshot timer was != cycle.  Don't we need to do the calibration + math crap we had with the hpet?